src/map_field.cpp

```diff
@@ -1335,17 +1337,18 @@ bool map::process_fire_field_in_submap( maptile &map_tile, const tripoint &p, fi
             // Allow weaker fires to spread occasionally
             const int power = cur.get_field_intensity() + one_in( 5 );
             if( can_spread && rng( 1, 100 - windpower ) < spread_chance &&
-                ( dster.is_flammable() || dsfrn.is_flammable() ) &&
+                ( dster.is_flammable() ||
+                  dsfrn.is_flammable() ||
+                  ( dst.get_item_count() > 0 &&
+                    flammable_items_at( p + eight_horizontal_neighbors[i] ) &&
```

Comment:
Please put call to `one_in( 5 )` before `flammable_items_at(...)` , as `flammable_items_at` is much more expensive than `one_in`. 

Also, could you please profile fire spreading in large area before and after the change? I'm a bit concerned about performance.

Reply:
Sure. I just wanted to change as little as possible from original code to avoid introducing any new problems. I will reorder the on_in( 5 ) though since it is part of the changed lines anyway.

I believe there might be other optimizations possibilities within process_fire_field_in_submap for example:
- reordering any && so likeliest to fail is first (see l1350-1352
- reordering any || so likeliest to succeed is first
- looking into whether flammable furniture/terrain has even to be checked again at that point since it was already given a chance to spread in a prior loop
- having nearwebfld next to terrain/furniture/flammable items check because I am increasingly certain that it should be there
...
However I would like to keep those potential optimizations separate from this PR since they might break something and need to be thoroughly tested.

Reply:
Please mark this as resolved.